Content.Shared/Automod/AutomodActionGroupPrototype.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Content.Shared.Database;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Automod;
+
+/// <summary>
+/// An automod filter for running actions when a pattern match is found in user input.
+/// </summary>
+/// <param name="pattern">The text to find in user input.</param>
+/// <param name="filterType">The type of filter to sort this definition into.</param>
+/// <param name="actionGroup">The <seealso cref="AutomodActionGroupPrototype"/> to run when the <paramref name="pattern"/> is matched</param>
+/// <param name="targets">The user input types that this filter applies to.</param>
+/// <param name="name">The user facing name of this filter.</param>
+public sealed class AutomodFilterDef(
+    int? id,
+    string pattern,
+    AutomodFilterType filterType,
+    string actionGroup,
+    AutomodTarget targets,
+    string name)
+{
+    public readonly int? Id = id;
+    public readonly string Pattern = pattern;
+    public readonly AutomodFilterType FilterType = filterType;
+    public readonly string ActionGroup = actionGroup;
+    public readonly AutomodTarget TargetFlags = targets;
+    public readonly string DisplayName = name;
+
+    public AutomodFilterDef(
+        string pattern,
+        AutomodFilterType filterType,
+        string actionGroup,
+        AutomodTarget targets,
+        string name) : this(null, pattern, filterType, actionGroup, targets, name)
+    {
+    }
+}
+
+/// <summary>
+/// A group of actions to be run when the automod pattern matches the text.
+/// </summary>
+[Prototype("automodActionGroup")]
+public sealed class AutomodActionGroupPrototype : IPrototype
+{
+    [ViewVariables]
+    [IdDataField]
+    public string ID { get; private set; } = default!;
+
+    [DataField]
+    public List<ITextAutomodAction> AutomodActions = new();
+}
+
+[ImplicitDataDefinitionForInheritors]
```

Comment:
Need someone to let me know if this is all that's needed or if the inheritors also need `[Serializable]` added.

Reply:
`Serializable` is completely unused by serv3